CADC SIA2 test failure

We started seeing this failure in the past week or so, seems to be more than transient now (but I haven't looked into the details yet, opening this issue to document the problem):

```
=================================== FAILURES ===================================
_____________________________ TestSIACadc.test_pos _____________________________

self = <[AttributeError("'Time' object has no attribute '_time'") raised in repr()] Time object at 0x7f5df1a91110>
val = array(nan), val2 = None, format = 'mjd', scale = None, precision = 3
in_subfmt = '*', out_subfmt = '*'

    def _get_time_fmt(self, val, val2, format, scale, precision, in_subfmt, out_subfmt):
        """
        Given the supplied val, val2, format and scale try to instantiate
        the corresponding TimeFormat class to convert the input values into
        the internal jd1 and jd2.
    
        If format is `None` and the input is a string-type or object array then
        guess available formats and stop when one matches.
        """
        if format is None and (
            val.dtype.kind in ("S", "U", "O", "M") or val.dtype.names
        ):
            # Input is a string, object, datetime, or a table-like ndarray
            # (structured array, recarray). These input types can be
            # uniquely identified by the format classes.
            formats = [
                (name, cls)
                for name, cls in self.FORMATS.items()
                if issubclass(cls, TimeUnique)
            ]
    
            # AstropyTime is a pseudo-format that isn't in the TIME_FORMATS registry,
            # but try to guess it at the end.
            formats.append(("astropy_time", TimeAstropyTime))
    
        elif not (isinstance(format, str) and format.lower() in self.FORMATS):
            if format is None:
                raise ValueError(
                    "No time format was given, and the input is not unique"
                )
            else:
                raise ValueError(
                    f"Format {format!r} is not one of the allowed formats "
                    f"{sorted(self.FORMATS)}"
                )
        else:
            formats = [(format, self.FORMATS[format])]
    
        assert formats
        problems = {}
        for name, cls in formats:
            try:
>               return cls(val, val2, scale, precision, in_subfmt, out_subfmt)

.tox/py311-test-alldeps-online/lib/python3.11/site-packages/astropy/time/core.py:597: 
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 
.tox/py311-test-alldeps-online/lib/python3.11/site-packages/astropy/time/formats.py:159: in __init__
    val1, val2 = self._check_val_type(val1, val2)
.tox/py311-test-alldeps-online/lib/python3.11/site-packages/astropy/time/formats.py:483: in _check_val_type
    val1, val2 = super()._check_val_type(val1, val2)
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 

self = <astropy.time.formats.TimeMJD object at 0x7f5df1e0f610>
val1 = array(nan), val2 = None

    def _check_val_type(self, val1, val2):
        """Input value validation, typically overridden by derived classes."""
        # val1 cannot contain nan, but val2 can contain nan
        isfinite1 = np.isfinite(val1)
        if val1.size > 1:  # Calling .all() on a scalar is surprisingly slow
            isfinite1 = (
                isfinite1.all()
            )  # Note: arr.all() about 3x faster than np.all(arr)
        elif val1.size == 0:
            isfinite1 = False
        ok1 = (
            val1.dtype.kind == "f"

    def _get_time_fmt(self, val, val2, format, scale, precision, in_subfmt, out_subfmt):
        """
        Given the supplied val, val2, format and scale try to instantiate
        the corresponding TimeFormat class to convert the input values into
        the internal jd1 and jd2.
    
        If format is `None` and the input is a string-type or object array then
        guess available formats and stop when one matches.
        """
        if format is None and (
            val.dtype.kind in ("S", "U", "O", "M") or val.dtype.names
        ):
            # Input is a string, object, datetime, or a table-like ndarray
            # (structured array, recarray). These input types can be
            # uniquely identified by the format classes.
            formats = [
                (name, cls)
                for name, cls in self.FORMATS.items()
                if issubclass(cls, TimeUnique)
            ]
    
            # AstropyTime is a pseudo-format that isn't in the TIME_FORMATS registry,
            # but try to guess it at the end.
            formats.append(("astropy_time", TimeAstropyTime))
    
        elif not (isinstance(format, str) and format.lower() in self.FORMATS):
            if format is None:
                raise ValueError(
                    "No time format was given, and the input is not unique"
                )
            else:
                raise ValueError(
                    f"Format {format!r} is not one of the allowed formats "
                    f"{sorted(self.FORMATS)}"
                )
        else:
            formats = [(format, self.FORMATS[format])]
    
        assert formats
        problems = {}
        for name, cls in formats:
            try:
                return cls(val, val2, scale, precision, in_subfmt, out_subfmt)
            except UnitConversionError:
                raise
            except (ValueError, TypeError) as err:
                # If ``format`` specified then there is only one possibility, so raise
                # immediately and include the upstream exception message to make it
                # easier for user to see what is wrong.
                if len(formats) == 1:
>                   raise ValueError(
                        f"Input values did not match the format class {format}:"
                        + os.linesep
                        + f"{err.__class__.__name__}: {err}"
                    ) from err
E                   ValueError: Input values did not match the format class mjd:
E                   TypeError: Input values for mjd class must be finite doubles

.tox/py311-test-alldeps-online/lib/python3.11/site-packages/astropy/time/core.py:605: ValueError
=========================== short test summary info ============================
XFAIL pyvo/io/vosi/tests/test_tables.py::TestTables::test_wrong_flag
FAILED pyvo/dal/tests/test_sia2_remote.py::TestSIACadc::test_pos - ValueError...
=== 1 failed, 318 passed, 1 xfailed, 1 xpassed, 1 rerun in 176.94s (0:02:56) ===
```

cc @andamian

--- a/CHANGES.rst
+++ b/CHANGES.rst
@@ -28,6 +28,9 @@
 
 - Fixed TapResults to inherit session. [#447]
 
+- Fix handling of nan values for Time properties in SIA2 records. [#463]
+
+
 1.4.1 (2023-03-07)
 ==================
 

--- a/pyvo/dal/sia2.py
+++ b/pyvo/dal/sia2.py
@@ -881,28 +881,30 @@ class ObsCoreRecord(SodaRecordMixin, DatalinkRecordMixin, Record,
     @property
     def t_min(self):
         """
-        The start time of the observation specified in MJD. In case of data
-        products result of the combination of multiple frames, min time must
-        be the minimum of the start times
+        The start time of the observation specified in MJD as an
+        `~astropy.time.Time` instance. In case of data products result of the
+        combination of multiple frames, min time must be the minimum of the
+        start times. ``None`` is used for NaN response values.
         """
         t_min = self.get('t_min')
         if np.isfinite(t_min):
             return Time(t_min, format='mjd')
         else:
-            return t_min
+            return None
 
     @property
     def t_max(self):
         """
-        The stop time of the observation specified in MJD. In case of data
-        products result of the combination of multiple frames, t_max must
-        be the maximum of the stop times
+        The stop time of the observation specified in MJD as an
+        `~astropy.time.Time` instance. In case of data products result of the
+        combination of multiple frames, t_max must be the maximum of the
+        stop times. ``None`` is used for NaN response values.
         """
         t_max = self.get('t_max')
         if np.isfinite(t_max):
             return Time(t_max, format='mjd')
         else:
-            return t_max
+            return None
 
     @property
     def t_exptime(self):

--- a/pyvo/dal/sia2.py
+++ b/pyvo/dal/sia2.py
@@ -11,6 +11,7 @@ endpoint.
 """
 
 import warnings
+import numpy as np
 
 from astropy import units as u
 from astropy.time import Time
@@ -884,7 +885,11 @@ class ObsCoreRecord(SodaRecordMixin, DatalinkRecordMixin, Record,
         products result of the combination of multiple frames, min time must
         be the minimum of the start times
         """
-        return Time(self.get('t_min'), format='mjd')
+        t_min = self.get('t_min')
+        if np.isfinite(t_min):
+            return Time(t_min, format='mjd')
+        else:
+            return t_min
 
     @property
     def t_max(self):
@@ -893,7 +898,11 @@ class ObsCoreRecord(SodaRecordMixin, DatalinkRecordMixin, Record,
         products result of the combination of multiple frames, t_max must
         be the maximum of the stop times
         """
-        return Time(self.get('t_min'), format='mjd')
+        t_max = self.get('t_max')
+        if np.isfinite(t_max):
+            return Time(t_max, format='mjd')
+        else:
+            return t_max
 
     @property
     def t_exptime(self):

